packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -96,91 +119,49 @@ export class SemanticViewerResultsRunner extends ResultsRunner {
         pivotLayoutOptions: VizPivotLayoutOptions[];
     } {
         return {
-            indexLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.DIMENSION) {
-                    acc.push({
-                        reference: column.reference,
-                        type: this.getAxisType(column),
-                    });
-                }
-                return acc;
-            }, [] as VizIndexLayoutOptions[]),
-            valuesLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.METRIC) {
-                    acc.push({
-                        reference: column.reference,
-                    });
-                }
-                return acc;
-            }, [] as VizValuesLayoutOptions[]),
-            pivotLayoutOptions: this.columns.filter((column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                return field?.kind === FieldType.DIMENSION;
-            }),
+            indexLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
+            valuesLayoutOptions: this.metrics.map((metric) => ({
+                reference: metric.name,
+            })),
+            pivotLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
         };
     }
 
     defaultPivotChartLayout(): VizChartLayout | undefined {
-        const xColumn = this.columns.find((column) => {
-            const field =
-                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                    this.fields,
-                    column.reference,
-                );
-            return field?.kind === FieldType.DIMENSION;
-        });
+        const xColumn = this.dimensions[0];
```

Comment:
Make this pick the first date dimension if it exists otherwise the first dimension